### PR TITLE
Advise newcomers to work on branch

### PR DESCRIPTION
**Pull request by Mariatta**
Add a paragraph explaining that they should not work on master branch.
Add instruction on how to checkout the branch.

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

Mention that newcomers should not work on master branch.
Add instruction on how to create a branch.

Closes https://github.com/coala/coala/issues/3148

### CODE REVIEW

File: docs/Developers/Newcomers_Guide.rst [hunk 1]
```diff
@@ -124,6 +124,14 @@ button from the website interface. To add it locally, simply run:
 where ``myfork`` is the name of your fork, and ``fork_link`` is a link to your
 fork repository.
 
+It is important that you do not make your changes on the master branch. To
+start working on an issue, you first need to create a new branch where you
+will work.
+
+::
+
+    $ git checkout -b branchname
```
Comment (Nosferatul) [inline docs/Developers/Newcomers_Guide.rst]: Maybe `<branchname>`?

**Comment by Nosferatul** [comment]
Also rebase needed

File: docs/Developers/Newcomers_Guide.rst [hunk 2]
```diff
@@ -124,6 +124,14 @@ button from the website interface. To add it locally, simply run:
 where ``myfork`` is the name of your fork, and ``fork_link`` is a link to your
 fork repository.
 
+It is important that you do not make your changes on the master branch. To
```
Comment (Mixih) [inline docs/Developers/Newcomers_Guide.rst]: Maybe put `.. note::` above here like the other notes so it shows up as a note?

**Comment by Mariatta** [comment]
Thanks, @Nosferatul and @Mixih 

Rebased, and adjusted based on your feedback.

**Comment by jayvdb** [comment]
A REALLY STRONG NOTE should also be added to "Step 6. Creating a Pull Request" . At this point, they should check they are not using "master", and advise them to go back to the previous step.

**Comment by Mixih** [comment]
>A REALLY STRONG NOTE

Maybe make it bold and italicized ? :smile:

**Comment by Mixih** [comment]
I think Jayvdb wanted a note around line 166 too asking them to check that they are on a branch before making a pull. Thanks!

**Comment by Mariatta** [comment]
Yes, I'm intending to do that but having trouble building docs now :( I posted my error with the docs build in gitter

**Comment by Mixih** [comment]
Ah yes, that would be jayvdb's commit.... Did you speak with him?

**Comment by jayvdb** [comment]
Continued on https://github.com/coala/coala/pull/3243

**Pull request closed without merging**